src/adapt/vgen.js

```diff
@@ -1205,6 +1210,59 @@ adapt.vgen.ViewFactory.prototype.preprocessElementStyle = function(computedStyle
 
 /**
  * @private
+ * @param {Element} element
+ * @param {adapt.cssstyler.AbstractStyler} styler
+ */
+adapt.vgen.ViewFactory.prototype.findAndProcessRepeatingElements = function(element, styler) {
+    for (var child = element.firstChild; child; child = child.nextSibling) {
+        if (child.nodeType !== 1) continue;
+        var computedStyle = {};
+        var elementStyle = styler.getStyle(/** @type {Element}*/ (child), false);
+        this.computeStyle(this.nodeContext.vertical, elementStyle, computedStyle);
+        var processRepeateOnBreak = this.processRepeateOnBreak(computedStyle);
+        if (!processRepeateOnBreak) continue;
+
+        if (this.nodeContext.formattingContext instanceof vivliostyle.repetitiveelements.RepetitiveElementsOwnerFormattingContext
+            && !this.nodeContext.formattingContext.isInherited(this.nodeContext)) {
+            return;
+        }
+
+        var parent = this.nodeContext.parent;
+        var parentFormattingContext = parent && parent.formattingContext;
+        this.nodeContext.formattingContext =
+            new vivliostyle.repetitiveelements.RepetitiveElementsOwnerFormattingContext(
+                parentFormattingContext, /** @type {!Element}*/ (this.nodeContext.sourceNode));
+        this.nodeContext.formattingContext.initializeRepetitiveElements(this.nodeContext.vertical);
+        return;
+    }
+};
+
+/**
+ * @private
+ * @param {!Object.<string,adapt.css.Val>} computedStyle
+ */
+adapt.vgen.ViewFactory.prototype.processRepeateOnBreak = function(computedStyle) {
+    var repeatOnBreak = computedStyle["repeat-on-break"];
+    if (repeatOnBreak !== adapt.css.ident.none) {
+        if (repeatOnBreak === adapt.css.ident.auto) {
+            if (this.nodeContext.display === "table-header-group") {
```

Comment:
When this function is called from `findAndProcessRepeatingElements` method (L1222), `this.nodeContext` points to the parent of the target element. Therefore it is wrong to use `this.nodeContext.display` value here.